ush/build.sh

```diff
@@ -20,6 +42,44 @@ REGRESSION_TESTS=${REGRESSION_TESTS:-"YES"} # Build regression test suite
 # Detect machine (sets MACHINE_ID)
 source $DIR_ROOT/ush/detect_machine.sh
 
+# checking if any specific option(s) is specified through the option arguments
+unset OPTS4CMAKE
+unset SAVELOG
+unset BUILD_GSDCLDANL
+while getopts "c:hlv" opt; do
+  case ${opt} in
+    c)
+      OPTS4CMAKE=${OPTARG}
+      ;;
+    v)
+      BUILD_VERBOSE=1
+      ;;
+    l)
+      SAVELOG="Yes"
+      ;;
+    h|\?|\:)
+      usage
+      ;;
+  esac
+done
+
+shift $((OPTIND - 1)) # Shift positional parameters to remove parsed options
+
+shopt -s nocasematch		# enable case-insensitive matching for patterns
+if [[ -v OPTS4CMAKE ]] ; then
+  case "${OPTS4CMAKE}" in
+    gsdcloud|gsdcldanl|gsdcld|gsdcloudanalysis|cloudanalysis|cldanl)
+      BUILD_GSDCLDANL="Yes"
+      echo " ****** Building GSI with GSD Cloud Analysis Support ****** "
+      ;;
+    *)
+      unset BUILD_GSDCLDANL
+      ;;
+  esac
+fi
+shopt -u nocasematch		# disable case-insensitive matching for patterns
+unset OPTS4CMAKE
```

Comment:
We use `OPTS4CMAKE` to set BUILD_GSDCLDANL`.  Why not have the `-c` option directly set this variable instead?  This seems more straightforward.